taskcluster/ci/go/kind.yml

```diff
@@ -2,30 +2,71 @@
 loader: taskgraph.loader.transform:loader
 
 transforms:
-  - src.transforms:taskcluster_image_versions
   - src.transforms:add_task_env
-  - src.transforms:podman_run
+  - src.transforms:parameterize_mounts
+  - src.transforms:parameterize_artifacts
   - taskgraph.transforms.job:transforms
   - taskgraph.transforms.task:transforms
 
 task-defaults:
   worker-type: gw-ubuntu-22-04
   run:
     using: bare
+    install:
+      - |
+        set -vx
+        export CGO_ENABLED=0
+        export GOROOT="$(pwd)/$GO_VERSION/go"
+        export GOPATH="$(pwd)/gopath$GO_RELEASE"
+        export PATH="${{GOPATH}}/bin:${{GOROOT}}/bin:$(pwd)/bin:${{PATH}}"
+        git --version
+        go version
+        go env
+        if [ ! -d taskcluster/.git ]; then
+          rm -rf taskcluster
+          git clone {head_repository} taskcluster
+        fi
+        cd taskcluster
+        git fetch {head_repository} "+${{GITHUB_SHA}}:refs/heads/X${{TASK_ID}}"
+        git checkout -f "X${{TASK_ID}}"
```

Comment:
just curious, do you remember how that `X{{TASK_ID}}` refs is created? Do we create a branch for each task, or previous `get fetch` is doing something I'm not aware of 🤔

Reply:
I'm not sure.. this is how other tasks do it as well so I just went along with the magic 😂

I think a new branch is created for each task though, yes.

Reply:
aaah.. that's the syntax to checkout specific SHA, yet create a branch (local ref) with this name.. neat